st2auth/st2auth/controllers/v1/auth.py

```diff
@@ -53,32 +49,41 @@ def post(self, request, **kwargs):
         except Exception:
             msg = 'Unexpected error occurred while verifying token.'
             LOG.exception(msg)
-            pecan.abort(http_client.INTERNAL_SERVER_ERROR, msg)
+            raise exc.HTTPInternalServerError(msg)
 
 
-class TokenController(rest.RestController):
+class TokenController(object):
     validate = TokenValidationController()
 
-    def __init__(self, *args, **kwargs):
-        super(TokenController, self).__init__(*args, **kwargs)
-
+    def __init__(self):
         try:
             self.handler = HANDLER_MAPPINGS[cfg.CONF.auth.mode]()
         except KeyError:
             raise ParamException("%s is not a valid auth mode" %
                                  cfg.CONF.auth.mode)
 
-    @jsexpose(body_cls=TokenAPI, status_code=http_client.CREATED)
     def post(self, request, **kwargs):
-        token = self.handler.handle_auth(request=request, headers=pecan.request.headers,
-                                         remote_addr=pecan.request.remote_addr,
-                                         remote_user=pecan.request.remote_user,
-                                         authorization=pecan.request.authorization,
+        headers = {}
+        if 'x-forwarded-for' in kwargs:
+            headers['x-forwarded-for'] = kwargs.pop('x-forwarded-for')
+
+        authorization = kwargs.pop('authorization', None)
+        if authorization:
+            authorization = tuple(authorization.split(' '))
```

Comment:
I don't think base64 encoded string can contain a whitespace, but maybe still safer to split on the first whitespace aka `.split(' ', 1)`.